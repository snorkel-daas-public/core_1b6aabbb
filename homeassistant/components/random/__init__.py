"""The random component."""

from homeassistant.config_entries import ConfigEntry
from homeassistant.core import HomeAssistant


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up a config entry."""
    await hass.config_entries.async_forward_entry_setups(
        entry, (entry.options["entity_type"],)
    )
    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(
        entry, (entry.options["entity_type"],)
    )
