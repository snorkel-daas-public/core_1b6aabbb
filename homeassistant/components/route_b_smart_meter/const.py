"""Constants for the Smart Meter B Route integration."""

from datetime import timedelta

DOMAIN = "route_b_smart_meter"
ENTRY_TITLE = "Route B Smart Meter"
DEFAULT_SCAN_INTERVAL = timedelta(seconds=300)

ATTR_API_INSTANTANEOUS_POWER = "instantaneous_power"
ATTR_API_TOTAL_CONSUMPTION = "total_consumption"
ATTR_API_INSTANTANEOUS_CURRENT_T_PHASE = "instantaneous_current_t_phase"
ATTR_API_INSTANTANEOUS_CURRENT_R_PHASE = "instantaneous_current_r_phase"
