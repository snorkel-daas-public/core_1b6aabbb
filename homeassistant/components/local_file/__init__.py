"""The local_file component."""

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_FILE_PATH, Platform
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ConfigEntryError

from .const import DOMAIN
from .util import check_file_path_access

PLATFORMS = [Platform.CAMERA]


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up Local file from a config entry."""
    file_path: str = entry.options[CONF_FILE_PATH]
    if not await hass.async_add_executor_job(check_file_path_access, file_path):
        raise ConfigEntryError(
            translation_domain=DOMAIN,
            translation_key="not_readable_path",
            translation_placeholders={"file_path": file_path},
        )

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload Local file config entry."""
    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
