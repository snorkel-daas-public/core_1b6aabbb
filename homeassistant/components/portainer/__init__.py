"""The Portainer integration."""

from __future__ import annotations

from pyportainer import Portainer

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import (
    CONF_API_KEY,
    CONF_API_TOKEN,
    CONF_HOST,
    CONF_URL,
    CONF_VERIFY_SSL,
    Platform,
)
from homeassistant.core import HomeAssistant
from homeassistant.helpers.aiohttp_client import async_create_clientsession

from .coordinator import PortainerCoordinator

_PLATFORMS: list[Platform] = [Platform.BINARY_SENSOR]

type PortainerConfigEntry = ConfigEntry[PortainerCoordinator]


async def async_setup_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Set up Portainer from a config entry."""

    client = Portainer(
        api_url=entry.data[CONF_URL],
        api_key=entry.data[CONF_API_TOKEN],
        session=async_create_clientsession(
            hass=hass, verify_ssl=entry.data[CONF_VERIFY_SSL]
        ),
    )

    coordinator = PortainerCoordinator(hass, entry, client)
    await coordinator.async_config_entry_first_refresh()

    entry.runtime_data = coordinator
    await hass.config_entries.async_forward_entry_setups(entry, _PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Unload a config entry."""
    return await hass.config_entries.async_unload_platforms(entry, _PLATFORMS)


async def async_migrate_entry(hass: HomeAssistant, entry: PortainerConfigEntry) -> bool:
    """Migrate old entry."""

    if entry.version < 2:
        data = dict(entry.data)
        data[CONF_URL] = data.pop(CONF_HOST)
        data[CONF_API_TOKEN] = data.pop(CONF_API_KEY)
        hass.config_entries.async_update_entry(entry=entry, data=data, version=2)

    return True
