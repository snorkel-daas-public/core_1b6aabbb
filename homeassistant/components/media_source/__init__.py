"""The media_source integration."""

from __future__ import annotations

from typing import Protocol

from homeassistant.components import websocket_api
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.integration_platform import (
    async_process_integration_platforms,
)
from homeassistant.helpers.typing import ConfigType

from . import http, local_source
from .const import (
    DOMAIN,
    MEDIA_CLASS_MAP,
    MEDIA_MIME_TYPES,
    MEDIA_SOURCE_DATA,
    URI_SCHEME,
    URI_SCHEME_REGEX,
)
from .error import MediaSourceError, Unresolvable
from .helper import async_browse_media, async_resolve_media
from .models import BrowseMediaSource, MediaSource, MediaSourceItem, PlayMedia

__all__ = [
    "DOMAIN",
    "MEDIA_CLASS_MAP",
    "MEDIA_MIME_TYPES",
    "BrowseMediaSource",
    "MediaSource",
    "MediaSourceError",
    "MediaSourceItem",
    "PlayMedia",
    "Unresolvable",
    "async_browse_media",
    "async_resolve_media",
    "generate_media_source_id",
    "is_media_source_id",
]


CONFIG_SCHEMA = cv.empty_config_schema(DOMAIN)


class MediaSourceProtocol(Protocol):
    """Define the format of media_source platforms."""

    async def async_get_media_source(self, hass: HomeAssistant) -> MediaSource:
        """Set up media source."""


def is_media_source_id(media_content_id: str) -> bool:
    """Test if identifier is a media source."""
    return URI_SCHEME_REGEX.match(media_content_id) is not None


def generate_media_source_id(domain: str, identifier: str) -> str:
    """Generate a media source ID."""
    uri = f"{URI_SCHEME}{domain or ''}"
    if identifier:
        uri += f"/{identifier}"
    return uri


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up the media_source component."""
    hass.data[MEDIA_SOURCE_DATA] = {}
    http.async_setup(hass)

    # Local sources support
    await _process_media_source_platform(hass, DOMAIN, local_source)
    hass.http.register_view(local_source.UploadMediaView)
    websocket_api.async_register_command(hass, local_source.websocket_remove_media)

    await async_process_integration_platforms(
        hass, DOMAIN, _process_media_source_platform
    )
    return True


async def _process_media_source_platform(
    hass: HomeAssistant,
    domain: str,
    platform: MediaSourceProtocol,
) -> None:
    """Process a media source platform."""
    source = await platform.async_get_media_source(hass)
    hass.data[MEDIA_SOURCE_DATA][domain] = source
    if isinstance(source, local_source.LocalSource):
        hass.http.register_view(local_source.LocalMediaView(hass, source))
