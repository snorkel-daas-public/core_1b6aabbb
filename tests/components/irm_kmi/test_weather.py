"""Test for the weather entity of the IRM KMI integration."""

from unittest.mock import AsyncMock

import pytest
from syrupy.assertion import SnapshotAssertion

from homeassistant.components.weather import (
    DOMAIN as WEATHER_DOMAIN,
    SERVICE_GET_FORECASTS,
)
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
import homeassistant.helpers.entity_registry as er

from tests.common import MockConfigEntry, snapshot_platform


@pytest.mark.freeze_time("2023-12-28T15:30:00+01:00")
async def test_weather_nl(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_irm_kmi_api_nl: AsyncMock,
    snapshot: SnapshotAssertion,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test weather with forecast from the Netherland."""
    mock_config_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)


@pytest.mark.parametrize(
    "forecast_type",
    ["daily", "hourly"],
)
@pytest.mark.freeze_time("2025-09-22T15:30:00+01:00")
async def test_forecast_service(
    hass: HomeAssistant,
    snapshot: SnapshotAssertion,
    mock_irm_kmi_api_nl: AsyncMock,
    mock_config_entry: MockConfigEntry,
    forecast_type: str,
) -> None:
    """Test multiple forecast."""
    mock_config_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    response = await hass.services.async_call(
        WEATHER_DOMAIN,
        SERVICE_GET_FORECASTS,
        {
            ATTR_ENTITY_ID: "weather.home",
            "type": forecast_type,
        },
        blocking=True,
        return_response=True,
    )
    assert response == snapshot


@pytest.mark.freeze_time("2024-01-21T14:15:00+01:00")
@pytest.mark.parametrize(
    "forecast_type",
    ["daily", "hourly"],
)
async def test_weather_higher_temp_at_night(
    hass: HomeAssistant,
    mock_config_entry: MockConfigEntry,
    mock_irm_kmi_api_high_low_temp: AsyncMock,
    forecast_type: str,
) -> None:
    """Test that the templow is always lower than temperature, even when API returns the opposite."""
    # Test case for https://github.com/jdejaegh/irm-kmi-ha/issues/8
    mock_config_entry.add_to_hass(hass)

    await hass.config_entries.async_setup(mock_config_entry.entry_id)
    await hass.async_block_till_done()

    response = await hass.services.async_call(
        WEATHER_DOMAIN,
        SERVICE_GET_FORECASTS,
        {
            ATTR_ENTITY_ID: "weather.home",
            "type": forecast_type,
        },
        blocking=True,
        return_response=True,
    )
    for forecast in response["weather.home"]["forecast"]:
        assert (
            forecast.get("native_temperature") is None
            or forecast.get("native_templow") is None
            or forecast["native_temperature"] >= forecast["native_templow"]
        )
