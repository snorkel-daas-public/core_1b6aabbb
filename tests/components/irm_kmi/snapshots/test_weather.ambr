# serializer version: 1
# name: test_forecast_service[daily]
  dict({
    'weather.home': dict({
      'forecast': list([
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2023-12-28',
          'is_daytime': True,
          'precipitation': 0.1,
          'precipitation_probability': None,
          'sunrise': '2023-12-28T08:47:43+01:00',
          'sunset': '2023-12-28T16:34:06+01:00',
          'temperature': 11.0,
          'templow': 9.0,
          'text': '''
            Waarschuwingen 
            Vanavond zijn er in het noordwesten zware windstoten mogelijk van 75-90 km/uur (code geel).
            
            Vanochtend is het half bewolkt met in het noorden kans op een bui. De wind komt uit het zuidwesten en is matig tot vrij krachtig, aan de kust krachtig tot hard, windkracht 6-7, af en toe even stormachtig, windkracht 8. Aan de kust komen windstoten voor van ongeveer 80 km/uur.
            Vanmiddag is het half tot zwaar bewolkt met kans op een bui, vooral in het noorden en westen. De middagtemperatuur ligt rond 11°C. De wind komt uit het zuidwesten en is meestal vrij krachtig, aan de kust krachtig tot hard, windkracht 6-7, vooral later ook af en toe stormachtig, windkracht 8. Aan de kust zijn er windstoten tot ongeveer 80 km/uur.
            Vanavond zijn er buien, alleen in het zuidoosten is het overwegend droog. De wind komt uit het zuidwesten en is meestal vrij krachtig, aan de kust hard tot stormachtig, windkracht 7 tot 8. Vooral in het noordwesten zijn windstoten mogelijk van 75-90 km/uur.
            
            Komende nacht komen er enkele buien voor. Met een minimumtemperatuur van ongeveer 8°C is het zacht. De wind komt uit het zuidwesten en is matig tot vrij krachtig, aan zee krachtig tot hard, windkracht 6-7, met zware windstoten tot ongeveer 80 km/uur.
            
            Morgenochtend is het half tot zwaar bewolkt en zijn er enkele buien. De wind komt uit het zuidwesten en is matig tot vrij krachtig, aan zee krachtig hard, windkracht 6-7, met vooral in het noordwesten mogelijk zware windstoten tot ongeveer 80 km/uur.
            Morgenmiddag is er af en toe ruimte voor de zon en blijft het op de meeste plaatsen droog, alleen in het zuidoosten kan een enkele bui vallen. Met middagtemperaturen van ongeveer 10°C blijft het zacht. De wind uit het zuidwesten is matig tot vrij krachtig, aan zee krachtig tot hard, windkracht 6-7, met in het Waddengebied zware windstoten tot ongeveer 80 km/uur.
            Morgenavond is het half tot zwaar bewolkt met een enkele bui. De wind komt uit het zuidwesten en is meest matig, aan de kust krachtig tot hard, windkracht 6-7, boven de Wadden eerst stormachtig, windkracht 8. 
            (Bron: KNMI, 2023-12-28T06:56:00+01:00)
  
          ''',
          'wind_bearing': 225.0,
          'wind_gust_speed': 33.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2023-12-29',
          'is_daytime': True,
          'precipitation': 3.8,
          'precipitation_probability': None,
          'sunrise': '2023-12-29T08:47:48+01:00',
          'sunset': '2023-12-29T16:35:00+01:00',
          'temperature': 10.0,
          'text': '',
          'wind_bearing': 248.0,
          'wind_gust_speed': 28.0,
          'wind_speed': 26.0,
        }),
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2023-12-30',
          'is_daytime': True,
          'precipitation': 1.7,
          'precipitation_probability': None,
          'sunrise': '2023-12-30T08:47:49+01:00',
          'sunset': '2023-12-30T16:35:57+01:00',
          'temperature': 10.0,
          'templow': 5.0,
          'text': '',
          'wind_bearing': 225.0,
          'wind_gust_speed': 25.0,
          'wind_speed': 22.0,
        }),
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2023-12-31',
          'is_daytime': True,
          'precipitation': 4.2,
          'precipitation_probability': None,
          'sunrise': '2023-12-31T08:47:47+01:00',
          'sunset': '2023-12-31T16:36:56+01:00',
          'temperature': 9.0,
          'templow': 7.0,
          'text': '',
          'wind_bearing': 203.0,
          'wind_gust_speed': 31.0,
          'wind_speed': 30.0,
        }),
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2024-01-01',
          'is_daytime': True,
          'precipitation': 2.2,
          'precipitation_probability': None,
          'sunrise': '2024-01-01T08:47:42+01:00',
          'sunset': '2024-01-01T16:37:59+01:00',
          'temperature': 7.0,
          'templow': 5.0,
          'text': '',
          'wind_bearing': 225.0,
          'wind_gust_speed': 28.0,
          'wind_speed': 23.0,
        }),
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2024-01-02',
          'is_daytime': True,
          'precipitation': 1.4,
          'precipitation_probability': None,
          'sunrise': '2024-01-02T08:47:32+01:00',
          'sunset': '2024-01-02T16:39:04+01:00',
          'temperature': 6.0,
          'templow': 3.0,
          'text': '',
          'wind_bearing': 225.0,
          'wind_gust_speed': 16.0,
          'wind_speed': 15.0,
        }),
        dict({
          'condition': 'pouring',
          'condition_2': None,
          'condition_evol': <ConditionEvol.STABLE: 'stable'>,
          'datetime': '2024-01-03',
          'is_daytime': True,
          'precipitation': 1.0,
          'precipitation_probability': None,
          'sunrise': '2024-01-03T08:47:20+01:00',
          'sunset': '2024-01-03T16:40:12+01:00',
          'temperature': 6.0,
          'templow': 3.0,
          'text': '',
          'wind_bearing': 203.0,
          'wind_gust_speed': 14.0,
          'wind_speed': 13.0,
        }),
      ]),
    }),
  })
# ---
# name: test_forecast_service[hourly]
  dict({
    'weather.home': dict({
      'forecast': list([
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T15:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1008.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 33.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T16:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1008.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T17:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1007.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T18:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1007.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T19:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T20:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-22T21:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-22T22:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.7,
          'precipitation_probability': 70,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-22T23:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.1,
          'precipitation_probability': 10,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 37.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T00:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 20,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-23T01:00:00+02:00',
          'is_daytime': False,
          'precipitation': 1.9,
          'precipitation_probability': 80,
          'pressure': 1005.0,
          'temperature': 10.0,
          'wind_bearing': 225.0,
          'wind_speed': 31.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-23T02:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.6,
          'precipitation_probability': 70,
          'pressure': 1005.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 38.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T03:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T04:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 34.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T05:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 35.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T06:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 34.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T07:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T08:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 31.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T09:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 31.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T10:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T11:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 32.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T12:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 34.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T13:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 33.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T14:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 31.0,
        }),
        dict({
          'condition': 'sunny',
          'datetime': '2025-09-23T15:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 10.0,
          'wind_bearing': 248.0,
          'wind_speed': 28.0,
        }),
        dict({
          'condition': 'sunny',
          'datetime': '2025-09-23T16:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 9.0,
          'wind_bearing': 248.0,
          'wind_speed': 24.0,
        }),
        dict({
          'condition': 'clear-night',
          'datetime': '2025-09-23T17:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 8.0,
          'wind_bearing': 225.0,
          'wind_speed': 20.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T18:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 8.0,
          'wind_bearing': 225.0,
          'wind_speed': 18.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T19:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1005.0,
          'temperature': 8.0,
          'wind_bearing': 203.0,
          'wind_speed': 15.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-23T20:00:00+02:00',
          'is_daytime': False,
          'precipitation': 5.7,
          'precipitation_probability': 100,
          'pressure': 1005.0,
          'temperature': 8.0,
          'wind_bearing': 248.0,
          'wind_speed': 22.0,
        }),
        dict({
          'condition': 'pouring',
          'datetime': '2025-09-23T21:00:00+02:00',
          'is_daytime': False,
          'precipitation': 3.8,
          'precipitation_probability': 100,
          'pressure': 1006.0,
          'temperature': 7.0,
          'wind_bearing': 270.0,
          'wind_speed': 26.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-23T22:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1006.0,
          'temperature': 8.0,
          'wind_bearing': 248.0,
          'wind_speed': 24.0,
        }),
        dict({
          'condition': 'cloudy',
          'datetime': '2025-09-23T23:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1007.0,
          'temperature': 7.0,
          'wind_bearing': 248.0,
          'wind_speed': 22.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T00:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1008.0,
          'temperature': 8.0,
          'wind_bearing': 270.0,
          'wind_speed': 26.0,
        }),
        dict({
          'condition': 'clear-night',
          'datetime': '2025-09-24T01:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1007.0,
          'temperature': 7.0,
          'wind_bearing': 270.0,
          'wind_speed': 26.0,
        }),
        dict({
          'condition': 'clear-night',
          'datetime': '2025-09-24T02:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1008.0,
          'temperature': 7.0,
          'wind_bearing': 270.0,
          'wind_speed': 24.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T03:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1008.0,
          'temperature': 7.0,
          'wind_bearing': 270.0,
          'wind_speed': 24.0,
        }),
        dict({
          'condition': 'clear-night',
          'datetime': '2025-09-24T04:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1009.0,
          'temperature': 7.0,
          'wind_bearing': 248.0,
          'wind_speed': 23.0,
        }),
        dict({
          'condition': 'clear-night',
          'datetime': '2025-09-24T05:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1009.0,
          'temperature': 6.0,
          'wind_bearing': 248.0,
          'wind_speed': 23.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T06:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1009.0,
          'temperature': 6.0,
          'wind_bearing': 248.0,
          'wind_speed': 21.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T07:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1010.0,
          'temperature': 6.0,
          'wind_bearing': 248.0,
          'wind_speed': 20.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T08:00:00+02:00',
          'is_daytime': False,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1011.0,
          'temperature': 6.0,
          'wind_bearing': 248.0,
          'wind_speed': 17.0,
        }),
        dict({
          'condition': 'sunny',
          'datetime': '2025-09-24T09:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1011.0,
          'temperature': 6.0,
          'wind_bearing': 248.0,
          'wind_speed': 13.0,
        }),
        dict({
          'condition': 'partlycloudy',
          'datetime': '2025-09-24T10:00:00+02:00',
          'is_daytime': True,
          'precipitation': 0.0,
          'precipitation_probability': 0,
          'pressure': 1012.0,
          'temperature': 5.0,
          'wind_bearing': 225.0,
          'wind_speed': 12.0,
        }),
      ]),
    }),
  })
# ---
# name: test_weather_nl[weather.home-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'config_subentry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'weather',
    'entity_category': None,
    'entity_id': 'weather.home',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': None,
    'original_name': None,
    'platform': 'irm_kmi',
    'previous_unique_id': None,
    'suggested_object_id': None,
    'supported_features': <WeatherEntityFeature: 7>,
    'translation_key': None,
    'unique_id': 'city country',
    'unit_of_measurement': None,
  })
# ---
# name: test_weather_nl[weather.home-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Weather data from the Royal Meteorological Institute of Belgium meteo.be',
      'friendly_name': 'Home',
      'precipitation_unit': <UnitOfPrecipitationDepth.MILLIMETERS: 'mm'>,
      'pressure': 1008.0,
      'pressure_unit': <UnitOfPressure.HPA: 'hPa'>,
      'supported_features': <WeatherEntityFeature: 7>,
      'temperature': 11.0,
      'temperature_unit': <UnitOfTemperature.CELSIUS: '°C'>,
      'uv_index': 1,
      'visibility_unit': <UnitOfLength.KILOMETERS: 'km'>,
      'wind_bearing': 225.0,
      'wind_speed': 40.0,
      'wind_speed_unit': <UnitOfSpeed.KILOMETERS_PER_HOUR: 'km/h'>,
    }),
    'context': <ANY>,
    'entity_id': 'weather.home',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'cloudy',
  })
# ---
